cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -216,52 +212,66 @@ public void execute(Project project) {
                 suite.setTests(TesterinaUtils.getSingleExecutionTests(suite, singleExecTests));
             }
             suite.setReportRequired(report || coverage);
-            Collection<Path> dependencies = jarResolver.getJarFilePathsRequiredForTestExecution(moduleName);
+
             if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
                 out.println("\t" + module.document(module.documentIds().iterator().next()).name());
             } else {
                 out.println("\t" + module.moduleName().toString());
             }
-            writeToJson(suite, moduleTestCachePath);
-            int testResult;
+
+            String resolvedModuleName =
+                    module.isDefaultModule() ? moduleName.toString() : module.moduleName().moduleNamePart();
+            testSuiteMap.put(resolvedModuleName, suite);
+            moduleNamesList.add(resolvedModuleName);
+        }
+
+        writeToTestSuiteJson(testSuiteMap, testsCachePath);
```

Comment:
As discussed during the previous review, shall we pass the json string to the process builder and load the json from the command argument string?

Reply:
Removed Implementation to write test suite to json. Now passes json string instead